core/src/test/java/org/axonframework/eventsourcing/eventstore/EventUtilsTest.java

```diff
@@ -77,6 +77,13 @@ public void testDomainEventStream_lastSequenceNumberEqualToLastProcessedEntryAft
         assertEquals(Long.valueOf(1L), eventStream.getLastSequenceNumber());
     }
 
+    @Test(expected = NullPointerException.class)
+    public void testDomainEventStream_nullPointerExceptionOnEmptyEventStream() {
+        DomainEventStream eventStream = EventUtils.upcastAndDeserializeDomainEvents(Stream.empty(),
+            serializer, NoOpEventUpcaster.INSTANCE, false);
+        long lastSequenceNumber = eventStream.getLastSequenceNumber();
```

Comment:
It will cause exception here: https://github.com/AxonFramework/AxonFramework/blob/master/core/src/main/java/org/axonframework/eventsourcing/EventSourcedAggregate.java#L212